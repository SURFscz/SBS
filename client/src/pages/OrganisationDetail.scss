@import "../stylesheets/vars.scss";

.mod-organisation-detail {

    background-color: $background;
    padding: 0;

    .title {
        font-size: 22px;
        color: $dark-grey;
        max-width: $medium;
        margin: 0 auto;
        padding: 10px 0 5px 0;

        p {
            padding: 10px 0 10px 0;
        }

        a {
            display: inline-block;
            font-size: 16px;
            text-transform: uppercase;
            text-decoration: none;
            margin: 10px 0;
            color: $blue;

            &:hover {
                font-weight: bolder;
            }

            svg {
                margin-right: 10px;
            }
        }
    }

    .organisation-detail {
        max-width: $medium;
        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, .1);
        margin: 0 auto 15px auto;
        background-color: white;
        padding: 10px $medium-padding 30px $medium-padding;


        .error {
            color: $red;
            display: inline-block;
            font-size: 14px;
        }

        div.input-field {
            margin-top: 15px;
        }

        section.actions {
            margin: $medium-margin 0 0 0;

            a.delete {
                margin-left: 25px;
            }
        }
    }

    .members-search, .collaborations-search {
        background-color: white;
        padding: $medium-padding;
        max-width: $medium;
        border-radius: $br;
        margin: 0 auto 10px auto;
        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, .1);
        position: relative;

        div.search {
            display: flex;
            padding-bottom: 15px;
            position: relative;
            min-height: 40px;

            svg.fa-search {
                position: absolute;
                font-size: 16px;
                color: $blue;
                top: 12px;
                left: 22px;
            }

            input[type=text] {
                flex-grow: 2;
                border: 1px solid $lighter-grey;
                padding-left: 45px;
                border-radius: $br;
                font-size: 16px;

                &.with-button {
                    border-top-right-radius: 0;
                    border-bottom-right-radius: 0;
                }

                &[disabled] {
                    background-color: $light-grey;
                }
            }

            a.button.blue {
                border-top-left-radius: 0;
                border-bottom-left-radius: 0;
            }
        }
    }

    .invitations-container, .api-keys-container {
        background-color: white;
        padding: $medium-padding;
        max-width: $medium;
        border-radius: $br;
        margin: 0 auto 10px auto;
        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, .1);
        position: relative;
    }

    .api-keys-container {
        a.button {
            margin-top: 25px;
        }
    }

    table.members, table.invitations, table.api-keys, table.collaborations {
        width: 100%;

        th {
            text-align: left;
            cursor: pointer;
            padding: 5px;

            svg {
                margin-left: 15px;
                color: $lighter-grey;

                &.current, &.reverse {
                    color: $blue;
                }
            }
        }

        tbody {
            tr {
                &:nth-child(odd) {
                    background-color: $lightest-grey;
                }
            }
        }

        td {
            vertical-align: middle;
            padding: 5px;

            .tooltip-container {
                margin-left: 15px;
                color: $blue;
            }
        }
    }

    table.members {
        th.name {
            width: 15%;
        }

        th.email {
            width: 15%;
        }

        th.uid {
            width: 25%;
        }

        th.role {
            width: 15%;
        }

        td.role {
            padding-right: 25px;
        }

        th.since {
            width: 20%;
        }

        th.actions {
            width: 10%;
            cursor: default;
        }

        td.actions {
            text-align: right;

            svg {
                color: $red;
                cursor: pointer;
                margin-right: 10px;
            }
        }
    }

    table.invitations {
        th.invitee {
            width: 20%;
        }

        th.invitedBy {
            width: 35%;
        }

        th.expires {
            width: 20%;
        }

        th.message {
            width: 25%;
        }

        td.message span {
            display: inline-block;
            max-width: 150px;
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
        }

        tbody {
            tr {
                cursor: pointer;

                &:hover {
                    color: $blue;
                }
            }

            td {
                &.role {
                    padding-right: 25px;

                }

                &.actions {
                    width: 35px;
                    text-align: left;
                    color: $blue;
                }

            }
        }
    }

    table.api-keys {
        th {
            cursor: default;
        }

        th.secret {
            width: 25%;
        }

        th.description {
            width: 75%;
        }

        td.actions {
            text-align: right;

            svg {
                color: $red;
                cursor: pointer;
                margin-right: 10px;
            }
        }
    }

    table.collaborations {
        th, td {
            cursor: pointer;
            padding: 10px 5px;
        }

        tbody {
            tr:hover {
                color: $blue;
            }

        }

        th.actions {
            width: 10%;
            cursor: default;
        }

        td.actions {
            text-align: right;
            z-index: 9;

            svg {
                color: $red;
                cursor: pointer;
                margin-right: 10px;
            }
        }

        td.link {
            color: $blue;
        }
    }


}
