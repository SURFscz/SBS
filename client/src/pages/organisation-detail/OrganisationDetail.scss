@import "../../stylesheets/vars";

.mod-organisation-container {
    .org-attributes-after {
        display: inline-block;
        margin-top: 10px;

    }
}
