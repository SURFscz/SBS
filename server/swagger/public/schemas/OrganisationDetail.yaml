---
type: object
properties:
  id:
    type: number
    readOnly: true
    example: 999
  name:
    type: string
    example: "University of Harderwijk"
  short_name:
    type: string
    example: "uniharderwijk"
  description:
    type: string
    example: "University of Harderwijk"
  category:
    type: string
    description: "Type of organisation"
    example: "Research"
  collaboration_creation_allowed:
    type: boolean
    description: "If true members a collaboration request is approved automatically"
    example: true
  services_restricted:
    type: boolean
    description: "If true only services exempted from restriction can be used by collaborations in this organisation"
    example: true
  uuid4:
    type: string
    format: uuid
    readOnly: true
    description: "Unique generated uuid for the cached image"
    example: "12035254-E1F1-4190-9E60-EDAEE8DEE72A"
  logo:
    type: string
    descriotion: "URL for the logo"
    example: "http://sram.surf.nl/api/images/collaborations/b7f621e3-de98-4de4-aca0-1573f4eb8339"
  created_at:
    type: number
    description: "The date of the created at in epoch seconds"
    example: 1644015600
  collaborations_count:
    type: number
    description: "The number of collaborations"
    example: 2
  organisation_memberships_count:
    type: number
    description: "The number of memberships"
    example: 2
  on_boarding_msg:
    type: string
    description: "Markdown on-boarding message for new members"
    example: "We are using SURF Research Access management to provide access to these research tools:\n- Wiki\n- Cloud\n"
  identifier:
    type: string
    format: uuid
    description: "Unique generated identifier"
    example: "42de0064-cddc-4c36-9e19-c0fd6e782956"
  collaborations:
    type: array
    items:
      $ref: '/swagger/schemas/CollaborationCreateResponse.yaml'
  schac_home_organisations:
    type: array
    items:
      $ref: '/swagger/schemas/SchacHomeOrganisation.yaml'
