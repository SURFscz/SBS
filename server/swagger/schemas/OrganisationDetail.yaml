---
type: object
properties:
  id:
    type: number
    readOnly: true
    example: 999
  name:
    type: string
    example: "UVA"
  short_name:
    type: string
    example: "uva"
  description:
    type: string
    example: "UVA university"
  category:
    type: string
    description: "Type of organisation"
    example: "Research"
  collaboration_creation_allowed:
    type: boolean
    description: "If true members a collaboration request is approved automatically"
    example: true
  services_restricted:
    type: boolean
    description: "If true only platform administrators can add services to this organisation"
    example: true
  uuid4:
    type: string
    format: uuid
    readOnly: true
    description: "Unique generated uuid for the cached image"
    example: "12035254-E1F1-4190-9E60-EDAEE8DEE72A"
  logo:
    type: string
    descriotion: "URL for the logo"
    example: "http://localhost:8080/api/images/collaborations/b7f621e3-de98-4de4-aca0-1573f4eb8339"
  created_at:
    type: number
    description: "The date of the created at in epoch seconds"
    example: 1644015600
  collaborations_count:
    type: number
    description: "The number of collaborations"
    example: 2
  organisation_memberships_count:
    type: number
    description: "The number of memberships"
    example: 2
  on_boarding_msg:
    type: string
    description: "Markdown on-boarding message for new members"
    example: "We are using **SRAM** to provide access to the following research tools:\n- Wiki\n- Cloud\n"
  identifier:
    type: string
    format: uuid
    descriotion: "Unique generated identifier"
    example: "42de0064-cddc-4c36-9e19-c0fd6e782956"
  collaborations:
    type: array
    items:
      $ref: '/swagger/schemas/CollaborationCreateResponse.yaml'
  schac_home_organisations:
    type: array
    items:
      $ref: '/swagger/schemas/SchacHomeOrganisation.yaml'
