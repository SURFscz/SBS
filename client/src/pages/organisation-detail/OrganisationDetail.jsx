import React from "react";
import {
    deleteOrganisationMembership,
    organisationById,
    organisationInvitationAccept,
    organisationInvitationByHash
} from "../../api";
import "./OrganisationDetail.scss";
import I18n from "../../locale/I18n";
import {isEmpty, stopEvent} from "../../utils/Utils";
import Tabs from "../../components/tabs/Tabs";
import UnitHeader from "../../components/redesign/unit-header/UnitHeader";
import OrganisationAdmins from "../../components/redesign/organisation-admins/OrganisationAdmins";
import {AppStore} from "../../stores/AppStore";
import Collaborations from "../../components/redesign/collaborations/Collaborations";
import SpinnerField from "../../components/redesign/spinner-field/SpinnerField";
import ApiKeys from "../../components/redesign/api-keys/ApiKeys";
import CollaborationRequests from "../../components/redesign/collaboration-requests/CollaborationRequests";
import OrganisationWelcomeDialog from "../../components/organisation-welcome-dialog/OrganisationWelcomeDialog";
import {actionMenuUserRole, isUserAllowed, ROLES} from "../../utils/UserRole";
import {getParameterByName} from "../../utils/QueryParameters";
import {setFlash} from "../../utils/Flash";
import ConfirmationDialog from "../../components/confirmation-dialog/ConfirmationDialog";
import {socket, SUBSCRIPTION_ID_COOKIE_NAME} from "../../utils/SocketIO";
import Users from "../../components/redesign/users/Users";
import {ButtonType} from "@surfnet/sds";
import {isInvitationExpired} from "../../utils/Date";
import ServiceConnectionRequests from "../../components/redesign/service-connection-requests/ServiceConnectionRequests";
import OrganisationOverview from "../organisation-overview/OrganisationOverview";

class OrganisationDetail extends React.Component {

    constructor(props, context) {
        super(props, context);
        this.state = {
            invitation: null,
            organisation: {},
            loading: true,
            tab: "collaborations",
            tabs: [],
            firstTime: false,
            confirmationDialogOpen: false,
            confirmationDialogAction: () => true,
            cancelDialogAction: () => this.setState({confirmationDialogOpen: false}),
            confirmationQuestion: "",
            socketSubscribed: false
        };
    }

    componentWillUnmount = () => {
        AppStore.update(s => {
            s.objectRole = null;
            s.actions = [];
        });
        const params = this.props.match.params;
        if (params.id) {
            const organisation_id = parseInt(params.id, 10);
            [`organisation_${organisation_id}`, "service"].forEach(topic => {
                socket.then(s => s.off(topic));
            });
        }
    }

    componentDidMount = callBack => {
        const params = this.props.match.params;
        if (params.hash) {
            organisationInvitationByHash(params.hash).then(res => {
                const {config, user} = this.props;
                this.setState({
                    invitation: res,
                    organisation: res.organisation,
                    adminOfOrganisation: false,
                    managerOfOrganisation: true,
                    loading: false,
                    firstTime: true,
                    confirmationDialogOpen: false,
                    isInvitation: true,
                    tabs: this.getTabs(res.organisation, config, true, user)
                });
            }).catch(() => {
                this.props.history.push("/404")
            });
        } else if (params.id) {
            const organisation_id = parseInt(params.id, 10);
            organisationById(organisation_id)
                .then(json => {
                    const {user, config} = this.props;
                    const member = (user.organisation_memberships || [])
                        .find(membership => membership.organisation_id === json.id);
                    if (isEmpty(member) && !user.admin) {
                        this.props.history.push("/");
                        return;
                    }
                    const {socketSubscribed} = this.state;
                    if (!socketSubscribed) {
                        [`organisation_${organisation_id}`, "service"].forEach(topic => {
                            socket.then(s => s.on(topic, data => {
                                const subscriptionIdSessionStorage = sessionStorage.getItem(SUBSCRIPTION_ID_COOKIE_NAME);
                                if (subscriptionIdSessionStorage !== data.subscription_id) {
                                    this.props.refreshUser(() => this.componentDidMount());
                                }
                            }));
                        });
                        this.setState({socketSubscribed: true})
                    }
                    const adminOfOrganisation = json.organisation_memberships
                        .some(member => member.role === "admin" && member.user_id === user.id) || user.admin;
                    const managerOfOrganisation = json.organisation_memberships
                        .some(member => member.role === "manager" && member.user_id === user.id);

                    const tab = params.tab || this.state.tab;
                    const tabs = this.getTabs(json, config, false, user);
                    AppStore.update(s => {
                        s.breadcrumb.paths = [
                            {path: "/", value: I18n.t("breadcrumb.home")},
                            {value: I18n.t("breadcrumb.organisation", {name: json.name})}
                        ];
                        s.objectRole = actionMenuUserRole(user, json, null, null, true)
                    });
                    const firstTime = getParameterByName("first", window.location.search) === "true";
                    this.tabChanged(tab, json.id);
                    this.setState({
                        organisation: json,
                        adminOfOrganisation: adminOfOrganisation,
                        managerOfOrganisation: managerOfOrganisation,
                        tab: tab,
                        tabs: tabs,
                        firstTime: firstTime,
                        loading: false
                    }, callBack);

                }).catch(() => {
                this.props.history.push("/")
            });
        } else {
            this.props.history.push("/404");
        }
    };

    onBoarding = () => {
        this.setState({firstTime: true});
    }

    getTabs = (organisation, config, isInvite, user) => {
        const isAdmin = isUserAllowed(ROLES.ORG_ADMIN, user, organisation.id);
        const tabs = isInvite ? [
            this.getCollaborationsTab(organisation),
        ] : [
            this.getCollaborationsTab(organisation),
            this.getCollaborationRequestsTab(organisation),
            this.getOrganisationAdminsTab(organisation, user),
            this.getServiceConnectionRequestsTab(organisation),
            config.api_keys_enabled ? this.getAPIKeysTab(organisation, user) : null,
            this.getUsersTab(organisation),
            isAdmin ? this.getOrganisationOverviewTab(organisation) : null
        ];
        return tabs.filter(tab => tab !== null);
    }

    getUsersTab = organisation => {
        return (<div key="users" name="users"
                     label={I18n.t("home.tabs.orgUsers")}>
            <Users {...this.props}
                   organisation={organisation}
                   adminSearch={false}/>
        </div>)
    }

    getOrganisationAdminsTab = (organisation, user) => {
        const expiredInvitations = (organisation.organisation_invitations || []).some(inv => isInvitationExpired(inv));
        const isOrgAdmin = isUserAllowed(ROLES.ORG_ADMIN, user, organisation.id);
        return (<div key="admins" name="admins"
                     label={I18n.t("home.tabs.orgAdmins", {count: organisation.organisation_memberships.length})}
                     notifier={expiredInvitations && isOrgAdmin}>
            <OrganisationAdmins {...this.props} organisation={organisation}
                                refresh={callback => this.componentDidMount(callback)}/>
        </div>)
    }

    getAPIKeysTab = (organisation, user) => {
        return (<div key="keys" name="keys"
                     label={I18n.t("home.tabs.apikeys", {count: (organisation.api_keys || []).length})}>
            <ApiKeys {...this.props} organisation={organisation} user={user}
                     refresh={callback => this.componentDidMount(callback)}/>
        </div>);
    }

    getOrganisationOverviewTab = organisation => {
        return (
            <div key="overview" name="overview"
                 label={I18n.t("home.tabs.orgOverview")}>
                <OrganisationOverview {...this.props}
                                      refresh={callback => this.componentDidMount(callback)}
                                      organisation={organisation}/>
            </div>
        );

    }

    getCollaborationsTab = organisation => {
        return (<div key="collaborations" name="collaborations"
                     label={I18n.t("home.tabs.orgCollaborations", {count: (organisation.collaborations || []).length})}>
            <Collaborations {...this.props}
                            collaborations={organisation.collaborations}
                            organisation={organisation}/>
        </div>)
    }

    getCollaborationRequestsTab = organisation => {
        const crl = (organisation.collaboration_requests || []).filter(cr => cr.status === "open").length;
        const tabLabel = I18n.t("home.tabs.collaborationRequests", {count: (organisation.collaboration_requests || []).length});
        return (<div key="collaboration_requests" name="collaboration_requests"
                     label={tabLabel}
                     notifier={crl > 0 ? crl : null}>
            <CollaborationRequests {...this.props} organisation={organisation}/>
        </div>)

    }

    getServiceConnectionRequestsTab = organisation => {
        const serviceConnectionRequests = organisation.collaborations
            .map(coll => {
                coll.service_connection_requests.forEach(scr => scr.collaboration = coll)
                return coll.service_connection_requests
            })
            .flat()
            .filter(scr => scr.pending_organisation_approval);

        if (serviceConnectionRequests.length === 0) {
            return null;
        }
        const nbr = serviceConnectionRequests.filter(scr => scr.status === "open").length;
        return (
            <div key="serviceConnectionRequests"
                 name="serviceConnectionRequests"
                 label={I18n.t("home.tabs.serviceConnectionRequests")}
                 notifier={nbr > 0 ? nbr : null}>
                <ServiceConnectionRequests
                    refresh={callback => this.componentDidMount(callback)}
                    serviceConnectionRequests={serviceConnectionRequests}
                    modelName={"serviceConnectionRequests"}
                    {...this.props} />
            </div>);
    }

    tabChanged = (name, id) => {
        const orgId = id || this.state.organisation.id;
        this.setState({tab: name}, () =>
            this.props.history.replace(`/organisations/${orgId}/${name}`));
    }

    doAcceptInvitation = () => {
        const {invitation, isInvitation} = this.state;
        if (isInvitation) {
            organisationInvitationAccept(invitation).then(() => {
                this.props.refreshUser(() => {
                    const path = encodeURIComponent(`/organisations/${invitation.organisation_id}`);
                    this.props.history.push(`/refresh-route/${path}`);
                });
            }).catch(e => {
                if (e.response && e.response.json) {
                    e.response.json().then(res => {
                        if (res.message && res.message.indexOf("already a member") > -1) {
                            this.setState({errorOccurred: true, firstTime: false}, () =>
                                setFlash(I18n.t("organisationInvitation.flash.alreadyMember"), "error"));
                        }
                    });
                } else {
                    throw e;
                }
            });
        } else {
            this.setState({firstTime: false});
        }
    }

    doDeleteMe = () => {
        this.setState({confirmationDialogOpen: false, loading: true});
        const {user} = this.props;
        const {organisation} = this.state;
        deleteOrganisationMembership(organisation.id, user.id)
            .then(() => {
                this.props.refreshUser(() => {
                    this.setState({confirmationDialogOpen: false, loading: false});
                    setFlash(I18n.t("organisationDetail.flash.memberDeleted", {name: user.name}));
                    if (user.admin) {
                        this.componentDidMount();
                    } else {
                        this.props.history.push("/home");
                    }
                });
            });
    };

    deleteMe = e => {
        stopEvent(e);
        this.setState({
            confirmationDialogOpen: true,
            confirmationQuestion: I18n.t("organisationDetail.deleteYourselfMemberConfirmation"),
            confirmationDialogAction: this.doDeleteMe
        });
    };

    getActions = (user, organisation, adminOfOrganisation) => {
        const actions = [];
        const isMember = organisation.organisation_memberships.find(m => m.user.id === user.id);
        const lastAdmin = adminOfOrganisation && organisation.organisation_memberships.filter(m => m.role === "admin").length < 2;
        if (isMember && (!lastAdmin || isMember.role !== "admin")) {
            actions.push({
                buttonType: ButtonType.DestructiveSecondary,
                name: I18n.t("models.organisations.leave"),
                perform: this.deleteMe
            });
        }
        if (!user.admin) {
            const queryParam = `name=${encodeURIComponent(organisation.name)}&back=${encodeURIComponent(window.location.pathname)}`;
            actions.push({
                buttonType: ButtonType.Secondary,
                name: I18n.t("home.history"),
                perform: () => this.props.history.push(`/audit-logs/organisations/${organisation.id}?${queryParam}`)
            });
        }
        return actions;
    }

    render() {
        const {
            tabs, organisation, loading, tab, firstTime, adminOfOrganisation,
            confirmationDialogOpen, cancelDialogAction, confirmationDialogAction, confirmationQuestion,
            isInvitation, invitation
        } = this.state;
        const {user} = this.props;
        if (loading) {
            return <SpinnerField/>;
        }
        let role;
        if (isInvitation) {
            role = invitation.intended_role === "admin" ? ROLES.ORG_ADMIN : ROLES.ORG_MANAGER;
        } else {
            role = adminOfOrganisation ? ROLES.ORG_ADMIN : ROLES.ORG_MANAGER;
        }
        return (
            <div className="mod-organisation-container">
                <OrganisationWelcomeDialog name={organisation.name}
                                           isOpen={firstTime}
                                           role={role}
                                           organisation={organisation}
                                           collaboration={null}
                                           isAdmin={user.admin}
                                           isInvitation={isInvitation}
                                           close={this.doAcceptInvitation}/>
                <ConfirmationDialog isOpen={confirmationDialogOpen}
                                    cancel={cancelDialogAction}
                                    confirm={confirmationDialogAction}
                                    isWarning={true}
                                    question={confirmationQuestion}/>
                <UnitHeader obj={organisation}
                            mayEdit={true}
                            displayDescription={true}
                            history={user.admin ? this.props.history : null}
                            auditLogPath={`organisations/${organisation.id}`}
                            breadcrumbName={I18n.t("breadcrumb.organisation", {name: organisation.name})}
                            firstTime={user.admin ? this.onBoarding : undefined}
                            name={organisation.name}
                            actions={this.getActions(user, organisation, adminOfOrganisation)}>
                    {organisation.accepted_user_policy &&
                        <a target="_blank" rel="noopener noreferrer" href={organisation.accepted_user_policy}>
                            {I18n.t("aup.title")}
                        </a>
                    }
                </UnitHeader>
                <Tabs activeTab={tab} tabChanged={this.tabChanged}>
                    {tabs}
                </Tabs>
            </div>);
    }
}

export default OrganisationDetail;
