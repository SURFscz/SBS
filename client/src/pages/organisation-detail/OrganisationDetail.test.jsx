import React from "react";
import {render} from "@testing-library/react";
import OrganisationDetail from "./OrganisationDetail";

describe("OrganisationDetail", () => {
    it("matches snapshot with default props", () => {
        const {asFragment} = render(
            <OrganisationDetail
                match={{params: {id: "1"}}}
                open={true}
                onClose={() => {
                }}
                onUpdate={() => {
                }}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom title", () => {
        const {asFragment} = render(
            <OrganisationDetail
                match={{params: {id: "1"}}}
                open={true}
                onClose={() => {
                }}
                onUpdate={() => {
                }}
                title="Custom Title"
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom description", () => {
        const {asFragment} = render(
            <OrganisationDetail
                match={{params: {id: "1"}}}
                open={true}
                onClose={() => {
                }}
                onUpdate={() => {
                }}
                description="Custom Description"
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
