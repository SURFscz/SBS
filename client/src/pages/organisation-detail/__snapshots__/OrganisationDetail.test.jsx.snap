// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganisationDetail matches snapshot with custom description 1`] = `
<DocumentFragment>
  <div
    class="sds--loading-container "
  >
    <div
      class="sds--loading"
    >
      <span
        class="sds--loading--dot"
      />
      <span
        class="sds--loading--dot"
      />
      <span
        class="sds--loading--dot"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`OrganisationDetail matches snapshot with custom title 1`] = `
<DocumentFragment>
  <div
    class="sds--loading-container "
  >
    <div
      class="sds--loading"
    >
      <span
        class="sds--loading--dot"
      />
      <span
        class="sds--loading--dot"
      />
      <span
        class="sds--loading--dot"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`OrganisationDetail matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="sds--loading-container "
  >
    <div
      class="sds--loading"
    >
      <span
        class="sds--loading--dot"
      />
      <span
        class="sds--loading--dot"
      />
      <span
        class="sds--loading--dot"
      />
    </div>
  </div>
</DocumentFragment>
`;
